From 1a7633c66e78df0e5b695c59d6527c5de2ae7aaa Mon Sep 17 00:00:00 2001
From: Carlos Gallegos <40046747+CharlyZest@users.noreply.github.com>
Date: Wed, 1 Dec 2021 02:51:35 -0600
Subject: [PATCH] Added Colorizer 1.1.0


diff --git a/pom.xml b/pom.xml
index 7f293b9d..de47f1e0 100644
--- a/pom.xml
+++ b/pom.xml
@@ -93,8 +93,8 @@
         </repository>
 
         <repository>
-            <id>royalmind-snapshots</id>
-            <url>https://packages.royalmind.net/repository/maven-snapshots/</url>
+            <id>royalmind</id>
+            <url>https://packages.royalmind.net/repository/maven-releases/</url>
         </repository>
     </repositories>
 
@@ -133,9 +133,9 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <groupId>net.royalmind.minecraft.library</groupId>
+            <groupId>net.royalmind.library</groupId>
             <artifactId>Colorizer</artifactId>
-            <version>1.0.2-SNAPSHOT</version>
+            <version>1.1.0</version>
             <scope>compile</scope>
         </dependency>
     </dependencies>
diff --git a/proxy/pom.xml b/proxy/pom.xml
index f29f1558..202276f8 100644
--- a/proxy/pom.xml
+++ b/proxy/pom.xml
@@ -18,6 +18,13 @@
     <name>Advocatus-Proxy</name>
     <description>Proxy component of the Elastic Portal Suite</description>
 
+    <repositories>
+        <repository>
+            <id>jitpack.io</id>
+            <url>https://jitpack.io</url>
+        </repository>
+    </repositories>
+
     <properties>
         <maven.deploy.skip>true</maven.deploy.skip>
         <maven.javadoc.skip>true</maven.javadoc.skip>
@@ -150,6 +157,15 @@
             <scope>runtime</scope>
         </dependency>
         <!-- Waterfall end -->
+
+        <!-- Advocatus start -->
+        <dependency>
+            <groupId>com.github.simplix-softworks</groupId>
+            <artifactId>simplixstorage</artifactId>
+            <version>3.2.4</version>
+            <scope>compile</scope>
+        </dependency>
+        <!-- Advocatus end -->
     </dependencies>
 
     <build>
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/Advocatus.java b/proxy/src/main/java/net/royalmind/software/advocatus/Advocatus.java
new file mode 100644
index 00000000..75d05ece
--- /dev/null
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/Advocatus.java
@@ -0,0 +1,2 @@
+package net.royalmind.software.advocatus;public class Advocatus {
+}
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/config/ConfigValue.java b/proxy/src/main/java/net/royalmind/software/advocatus/config/ConfigValue.java
new file mode 100644
index 00000000..66edfe4d
--- /dev/null
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/config/ConfigValue.java
@@ -0,0 +1,2 @@
+package net.royalmind.software.advocatus.config;public @interface ConfigValue {
+}
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/config/Settings.java b/proxy/src/main/java/net/royalmind/software/advocatus/config/Settings.java
new file mode 100644
index 00000000..c5b5083a
--- /dev/null
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/config/Settings.java
@@ -0,0 +1,2 @@
+package net.royalmind.software.advocatus.config;public class Settings {
+}
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java b/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
new file mode 100644
index 00000000..900387b0
--- /dev/null
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
@@ -0,0 +1,2 @@
+package net.royalmind.software.advocatus.util;public class Logger {
+}
diff --git a/proxy/src/main/resources/messages-es.properties b/proxy/src/main/resources/messages-es.properties
new file mode 100644
index 00000000..e69de29b
-- 
2.33.0.windows.2

