From a794ebf76117914ad3128513b241e9183feaea1e Mon Sep 17 00:00:00 2001
From: Carlos Gallegos <40046747+CharlyZest@users.noreply.github.com>
Date: Wed, 1 Dec 2021 03:27:22 -0600
Subject: [PATCH] Separate package with sources of Advocatus.


diff --git a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
index 11a5cf21..a8120fe5 100644
--- a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
+++ b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
@@ -93,6 +93,7 @@ import net.md_5.bungee.protocol.packet.PluginMessage;
 import net.md_5.bungee.query.RemoteQuery;
 import net.md_5.bungee.scheduler.BungeeScheduler;
 import net.md_5.bungee.util.CaseInsensitiveMap;
+import net.royalmind.software.advocatus.Advocatus;
 
 /**
  * Main BungeeCord proxy class.
@@ -235,17 +236,17 @@ public class BungeeCord extends ProxyServer
         {
             if ( EncryptionUtil.nativeFactory.load() )
             {
-                logger.info( "Using mbed TLS based native cipher." );
+                logger.info( "Utilizando cifrado nativo basado en TLS mbed." );
             } else
             {
-                logger.info( "Using standard Java JCE cipher." );
+                logger.info( "Usando cifrado est√°ndar de Java JCE." );
             }
             if ( CompressFactory.zlib.load() )
             {
-                logger.info( "Using zlib based native compressor." );
+                logger.info( "Usando un compresor nativo basado en zlib." );
             } else
             {
-                logger.info( "Using standard Java compressor." );
+                logger.info( "Utilizando el compresor est√°ndar de Java." );
             }
         }
     }
@@ -341,7 +342,7 @@ public class BungeeCord extends ProxyServer
                     if ( future.isSuccess() )
                     {
                         listeners.add( future.channel() );
-                        getLogger().log( Level.INFO, "Listening on {0}", info.getSocketAddress() );
+                        getLogger().log( Level.INFO, "Listening on PORT {0}", info.getSocketAddress() );
                     } else
                     {
                         getLogger().log( Level.WARNING, "Could not bind to host " + info.getSocketAddress(), future.cause() );
@@ -537,10 +538,7 @@ public class BungeeCord extends ProxyServer
     }
 
     @Override
-    public String getName()
-    {
-        return "Advocatus";
-    }
+    public String getName() { return Advocatus.getInstance().getName(); }
 
     @Override
     public String getVersion()
@@ -550,33 +548,11 @@ public class BungeeCord extends ProxyServer
 
     public void reloadMessages()
     {
-        File file = new File( "messages.properties" );
-        if ( file.isFile() )
-        {
-            try ( FileReader rd = new FileReader( file ) )
-            {
-                customBundle = new PropertyResourceBundle( rd );
-            } catch ( IOException ex )
-            {
-                getLogger().log( Level.SEVERE, "Could not load custom messages.properties", ex );
-            }
-        }
+        Advocatus.getInstance().loadCustomBundle();
     }
 
     @Override
-    public String getTranslation(String name, Object... args)
-    {
-        String translation = "<translation '" + name + "' missing>";
-        try
-        {
-            final String string = customBundle != null && customBundle.containsKey( name ) ? customBundle.getString( name ) : baseBundle.getString( name );
-
-            translation = ( args.length == 0 ) ? string : MessageFormat.format( string, args );
-        } catch ( MissingResourceException ex )
-        {
-        }
-        return translation;
-    }
+    public String getTranslation(String name, Object... args) { return Advocatus.getInstance().getTranslation(name, args); }
 
     @Override
     @SuppressWarnings("unchecked")
@@ -813,4 +789,10 @@ public class BungeeCord extends ProxyServer
     {
         return new BungeeTitle();
     }
+
+    /* Advocatus Start */
+
+    public ResourceBundle getBaseBundle() { return baseBundle; }
+
+    /* Advocatus End */
 }
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
index d3650ff8..71076f3f 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
@@ -26,6 +26,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Objects; // Waterfall
 import lombok.RequiredArgsConstructor;
+import net.md_5.bungee.BungeeCord;
 import net.md_5.bungee.ServerConnection;
 import net.md_5.bungee.ServerConnection.KeepAliveData;
 import net.md_5.bungee.UserConnection;
@@ -294,7 +295,7 @@ public class DownstreamBridge extends PacketHandler
             Preconditions.checkState( !serverBrand.contains( bungee.getName() ), "Cannot connect proxy to itself!" );
 
             brand = ByteBufAllocator.DEFAULT.heapBuffer();
-            DefinedPacket.writeString( bungee.getName() + " <- " + serverBrand, brand ); // Waterfall
+            DefinedPacket.writeString(BungeeCord.getInstance().getTranslation("brand-name"), brand ); // Advocatus
             pluginMessage.setData( brand );
             brand.release();
             // changes in the packet are ignored so we need to send it manually
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/Advocatus.java b/proxy/src/main/java/net/royalmind/software/advocatus/Advocatus.java
index 75d05ece..42f2cbd7 100644
--- a/proxy/src/main/java/net/royalmind/software/advocatus/Advocatus.java
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/Advocatus.java
@@ -1,2 +1,97 @@
-package net.royalmind.software.advocatus;public class Advocatus {
+package net.royalmind.software.advocatus;
+
+import java.io.File;
+import java.text.MessageFormat;
+import java.util.*;
+
+import net.royalmind.library.colorizer.Colorizer;
+import net.royalmind.software.advocatus.util.Logger;
+import de.leonhard.storage.Yaml;
+
+public class Advocatus
+{
+    private static Advocatus INSTANCE;
+    private final String FORK_NAME = "Advocatus";
+
+    private ResourceBundle baseMessages;
+    private Yaml customMessages;
+    public final String baseConfigDir = FORK_NAME + "/";
+    private final HashMap<String, String> MESSAGES_CACHE;
+
+    public Advocatus()
+    {
+        INSTANCE = this;
+        MESSAGES_CACHE = new HashMap<>();
+        loadBaseBundle();
+    }
+
+    private void loadBaseBundle()
+    {
+        try
+        { baseMessages = ResourceBundle.getBundle("messages-es"); }
+        catch ( MissingResourceException ex )
+        { baseMessages = ResourceBundle.getBundle( "messages", Locale.ENGLISH ); }
+    }
+
+    public void loadCustomBundle()
+    {
+        final String customMessagesFileName = "messages-es";
+        try
+        {
+            File file = new File( baseConfigDir, customMessagesFileName );
+
+            if (!file.exists())
+            {
+                file.mkdir();
+
+                customMessages = new Yaml(file);
+                final Enumeration<String> keys = baseMessages.getKeys();
+
+                while (keys.hasMoreElements())
+                {
+                    final String key = keys.nextElement();
+                    if (customMessages.get(key, "").isEmpty()) customMessages.set(key, baseMessages.getString(key));
+                }
+            }
+            else
+            {
+                customMessages = new Yaml(file);
+                final Enumeration<String> keys = baseMessages.getKeys();
+
+                while (keys.hasMoreElements())
+                {
+                    final String key = keys.nextElement();
+                    if (customMessages.get(key, "").isEmpty()) customMessages.set(key, baseMessages.getString(key));
+                }
+            }
+
+            MESSAGES_CACHE.clear();
+        }
+        catch (Exception ex) { ex.printStackTrace(); }
+    }
+
+    public String getTranslation(String name, Object... args)
+    {
+        String translation = "&c<translation '" + name + "' missing>&r";
+        try
+        {
+            if (!MESSAGES_CACHE.containsKey(name))
+            {
+                final String VALUE = customMessages != null && customMessages.contains( name )
+                        ? customMessages.getString( name )
+                        : baseMessages.getString( name );
+
+                translation = Colorizer.translate(( args.length == 0 ) ? VALUE : MessageFormat.format( VALUE, args ));
+                MESSAGES_CACHE.put(name, translation);
+            }
+            else { translation = MESSAGES_CACHE.get(name); }
+        }
+        catch ( MissingResourceException ex ) { Logger.error("Fallo al cargar la traduccion de: \""+ name +"\"", ex); }
+
+        return translation;
+    }
+
+    public String getName() { return FORK_NAME; }
+
+    public static Advocatus getInstance() { return INSTANCE == null ? new Advocatus() : INSTANCE; }
 }
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/config/ConfigValue.java b/proxy/src/main/java/net/royalmind/software/advocatus/config/ConfigValue.java
index 66edfe4d..6896de8a 100644
--- a/proxy/src/main/java/net/royalmind/software/advocatus/config/ConfigValue.java
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/config/ConfigValue.java
@@ -1,2 +1,13 @@
-package net.royalmind.software.advocatus.config;public @interface ConfigValue {
+package net.royalmind.software.advocatus.config;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ConfigValue
+{
+    String value();
 }
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/config/Settings.java b/proxy/src/main/java/net/royalmind/software/advocatus/config/Settings.java
index c5b5083a..cfbba315 100644
--- a/proxy/src/main/java/net/royalmind/software/advocatus/config/Settings.java
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/config/Settings.java
@@ -1,2 +1,16 @@
-package net.royalmind.software.advocatus.config;public class Settings {
+package net.royalmind.software.advocatus.config;
+
+public class Settings
+{
+    @ConfigValue("version")
+    public static int VERSION;
+
+    @ConfigValue("debug")
+    public static boolean DEBUGMODE;
+
+    static
+    {
+        Settings.VERSION = 1;
+        Settings.DEBUGMODE = true;
+    }
 }
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java b/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
index 900387b0..1fc6d22a 100644
--- a/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
@@ -1,2 +1,97 @@
-package net.royalmind.software.advocatus.util;public class Logger {
+package net.royalmind.software.advocatus.util;
+
+import java.util.logging.Level;
+
+import net.md_5.bungee.BungeeCord;
+import net.royalmind.library.colorizer.Colorizer;
+import net.royalmind.software.advocatus.config.Settings;
+
+public class Logger
+{
+    public static java.util.logging.Logger logger;
+
+    /*
+     * Registra un mensaje de informaci√≥n en el log.
+     * @param texto Mensaje informativo.
+     */
+    public static void info(final String texto, final Object... array)
+    {
+        final String[] split = String.format(texto, array).split("\n");
+
+        for (int length = split.length, i = 0; i < length; ++i)
+        {
+            Logger.logger.info(Colorizer.translate("" + split[i]));
+        }
+    }
+
+    /*
+     * Registra un mensaje de advertencia en el log.
+     * @param texto Mensaje informativo.
+     */
+    public static void warn(final String texto, final Object... array)
+    {
+        final String[] split = String.format(texto, array).split("\n");
+
+        for (int length = split.length, i = 0; i < length; ++i)
+        {
+            Logger.logger.warning(Colorizer.translate("" + split[i]));
+        }
+    }
+
+    /*
+     * Registra un mensaje de error grave en el log.
+     * @param texto Mensaje informativo.
+     */
+    public static void severe(final String texto, final Object... array)
+    {
+        final String[] split = String.format(texto, array).split("\n");
+
+        for (int length = split.length, i = 0; i < length; ++i) {
+            Logger.logger.severe(Colorizer.translate("" + split[i]));
+        }
+    }
+
+    /*
+     * Registra un mensaje de error en el log.
+     * @param texto Mensaje informativo.
+     */
+    public static void error(final String texto, final Object... array)
+    {
+        final String[] split = String.format(texto, array).split("\n");
+
+        for (int length = split.length, i = 0; i < length; ++i)
+        {
+            info("&c[ERROR] &r%s", split[i]);
+        }
+    }
+    /*
+     * Registra un mensaje de error en el log.
+     * @param texto Mensaje informativo.
+     */
+    public static void error(final String texto, final Exception exception)
+    {
+        logger.log(Level.SEVERE, Colorizer.translate("&c[ERROR] &r" + texto + "\t&cDetalle: " + exception.toString()));
+    }
+
+    /*
+     * Registra un mensaje de depuraci√≥n en el log.
+     * @param texto Mensaje informativo.
+     */
+    public static void debug(final String texto, final Object... array)
+    {
+        if (Settings.DEBUGMODE) { info(String.format(texto, array), new Object[0]); }
+    }
+
+    /*
+     * Registra un mensaje de depuraci√≥n en el log.
+     * @param texto Mensaje informativo.
+     */
+    public static void debugError(final String texto, final Object... array)
+    {
+        if (Settings.DEBUGMODE) { error(String.format(texto, array), new Object[0]); }
+    }
+
+    public static String getLine() { return Colorizer.translate("&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a="); }
+
+    static { Logger.logger = BungeeCord.getInstance().getLogger(); }
 }
diff --git a/proxy/src/main/resources/messages-es.properties b/proxy/src/main/resources/messages-es.properties
index e69de29b..d4d3132a 100644
--- a/proxy/src/main/resources/messages-es.properties
+++ b/proxy/src/main/resources/messages-es.properties
@@ -0,0 +1,40 @@
+alert=&8[&4Anuncio&8]&r 
+already_connected=&c°Ya esta conectado al servidor!
+already_connected_proxy=&cYa estas conectado a este proxy!
+already_connecting=&c°Ya esta conectando a este servidor!
+command_list=&a[{0}] &e({1})=&r{2}
+connect_kick=&cNo se pudo conectar a &e{0}&c={1}
+current_server=&6Actualmente estas conectado a {0}.
+fallback_kick=&cNo se pudo conectar a un servidor predeterminado o alternativo. Vuelve a intentarlo mas tarde={0}
+fallback_lobby=&cNo se pudo conectar al servidor de destino, se le ha trasladado a un servidor de reserva.
+timeout=&eServidor no accesible (tiempo de espera). øDesconectado? øDireccion/puerto/cortafuegos configurados incorrectamente?
+lost_connection=&r[&e&lRoyal&6&lMind &f&lNetwork&r] Conexion perdida con nuestros servidores.
+mojang_fail=Se produjo un error al contactar con los servidores de inicio de sesion de Mojang, øEstan inactivos? :(
+no_permission=&c°No tienes permiso para ejecutar este comando!
+no_server=&cEl servidor especificado no existe.
+no_server_permission=&cNo tienes permiso para acceder a este servidor.
+outdated_client=Cliente desactualizado! Por favor usa la version {0}
+outdated_server=°Servidor desactualizado! Todavia estoy en {0}
+proxy_full=°El servidor esta lleno!
+restart=&r[&e&lRoyal&6&lMind &f&lNetwork&r] Reiniciando network
+server_list=&6Puede conectarse a los siguientes servidores en este momento=
+server_went_down=&cEl servidor en el que estaba anteriormente se cayo, se ha conectado a un servidor de reserva
+total_players=Total de jugadores en linea={0}
+name_invalid=Nombre de usuario contiene caracteres invalidos.
+ping_cannot_connect=&r[&e&lRoyal&6&lMind &f&lNetwork&r] &cNo se puede conectar al servidor.
+offline_mode_player=Cuenta no autenticada con Minecraft.net
+message_needed=&cDebes proporcionar un mensaje.
+error_occurred_player=&cSe produjo un error al analizar su mensaje. (Coloca el cursor sobre los detalles)
+error_occurred_console=&cSe produjo un error al analizar su mensaje={0}
+click_to_connect=Haga clic para conectarse al servidor
+username_needed=&cSiga este comando con un nombre de usuario.
+user_not_online=&cEse usuario no esta en linea.
+user_online_at=&a{0} &resta en linea en {1}
+send_cmd_usage=&cNo hay suficientes argumentos, uso=/send <servidor|jugador|all|current> <servidor objetivo>
+player_only=&cSolo en el juego los jugadores pueden usar este comando.
+you_got_summoned=&6Transportado a {0} por {1}
+command_perms_groups=&6Tienes los siguientes grupos={0}
+command_perms_permission=&9- {0}
+command_ip=&9La IP de {0} es {1}
+illegal_chat_characters=&cCaracteres ilegales en el chat ({0})
+brand-name=&e&oplay.royalmind.net
\ No newline at end of file
-- 
2.33.0.windows.2

