From 7ec4f2cb30fcb60afedbb884b49a7c205d5e3a67 Mon Sep 17 00:00:00 2001
From: Carlos Gallegos <40046747+CharlyZest@users.noreply.github.com>
Date: Tue, 11 Jan 2022 01:02:25 -0600
Subject: [PATCH] Updated pom.xml


diff --git a/pom.xml b/pom.xml
index 9989601f..d16e74a2 100644
--- a/pom.xml
+++ b/pom.xml
@@ -69,12 +69,12 @@
     <distributionManagement>
         <repository>
             <id>royalmind-releases</id>
-            <url>https://packages.royalmind.net/repository/maven-releases/</url>
+            <url>https://repo.royalmind.net/repository/maven-releases/</url>
         </repository>
 
         <snapshotRepository>
             <id>royalmind-snapshots</id>
-            <url>https://packages.royalmind.net/repository/maven-snapshots/</url>
+            <url>https://repo.royalmind.net/repository/maven-snapshots/</url>
         </snapshotRepository>
     </distributionManagement>
 
@@ -93,8 +93,18 @@
         </repository>
 
         <repository>
-            <id>royalmind</id>
-            <url>https://packages.royalmind.net/repository/maven-releases/</url>
+            <id>exceptionflug</id>
+            <url>https://mvn.exceptionflug.de/repository/exceptionflug-public/</url>
+        </repository>
+
+        <repository>
+            <id>royalmind-releases</id>
+            <url>https://repo.royalmind.net/repository/maven-releases/</url>
+        </repository>
+
+        <repository>
+            <id>royalmind-snapshots</id>
+            <url>https://repo.royalmind.net/repository/maven-snapshots/</url>
         </repository>
     </repositories>
 
@@ -251,9 +261,9 @@
             </build>
         </profile>
         <profile>
-            <id>jdk-15-javadoc</id>
+            <id>jdk-16-javadoc</id>
             <activation>
-                <jdk>[15,)</jdk>
+                <jdk>[16,)</jdk>
             </activation>
             <build>
                 <pluginManagement>
diff --git a/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java b/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
index 54f9ebd5..9c11de11 100644
--- a/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
+++ b/proxy/src/main/java/net/royalmind/software/advocatus/util/Logger.java
@@ -20,7 +20,7 @@ public class Logger
 
         for (int length = split.length, i = 0; i < length; ++i)
         {
-            Logger.logger.info(Colorizer.transale("" + split[i]));
+            Logger.logger.info(Colorizer.decorate("" + split[i]));
         }
     }
 
@@ -34,7 +34,7 @@ public class Logger
 
         for (int length = split.length, i = 0; i < length; ++i)
         {
-            Logger.logger.warning(Colorizer.transale("" + split[i]));
+            Logger.logger.warning(Colorizer.decorate("" + split[i]));
         }
     }
 
@@ -47,7 +47,7 @@ public class Logger
         final String[] split = String.format(texto, array).split("\n");
 
         for (int length = split.length, i = 0; i < length; ++i) {
-            Logger.logger.severe(Colorizer.transale("" + split[i]));
+            Logger.logger.severe(Colorizer.decorate("" + split[i]));
         }
     }
 
@@ -70,7 +70,7 @@ public class Logger
      */
     public static void error(final String texto, final Exception exception)
     {
-        logger.log(Level.SEVERE, Colorizer.transale("&c[ERROR] &r" + texto + "\t&cDetalle: " + exception.toString()));
+        logger.log(Level.SEVERE, Colorizer.decorate("&c[ERROR] &r" + texto + "\t&cDetalle: " + exception.toString()));
     }
 
     /*
@@ -102,4 +102,4 @@ public class Logger
     public static String getLine() { return Colorizer.transale("&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a=&e-&a="); }
 
     static { Logger.logger = BungeeCord.getInstance().getLogger(); }
-}
+}
\ No newline at end of file
-- 
2.33.0.windows.2

